### PR TITLE
Clean up go to definition tests and adds a test for variable tokens

### PR DESCRIPTION
**Pull request by JoshuaBatty**
This PR cleans up the go_to testing code so it's easier to use. 

I've also added a new example so we can test the definition response of variables in the following cases:

- Variable expressions
- Function arguments
- Struct fields
- Enum fields
- Tuple elements
- Array elements
- Scoped declarations
- If let scopes
- Shadowing

### CODE REVIEW

File: sway-lsp/src/server.rs [hunk 1]
```diff
@@ -1302,19 +1264,59 @@ mod tests {
...
+            req_uri: &uri,
+            req_line: 15,
+            req_char: 12,
+            def_line: 80,
+            def_start_char: 9,
+            def_end_char: 15,
+            def_path: "sway-lib-std/src/option.sw",
+        };
+
+        let _ = definition_check(&mut service, &opt_go_to, 1).await;
```
Comment (sdankel) [inline sway-lsp/src/server.rs:1277]: nit to make the test cases more readable:

```
        fn get_go_to_test_case(
            base: GotoDefinition,
            req_line: u64,
            req_char: u64,
        ) -> GotoDefintion {
            GotoDefintion {
                ..base,
                req_line,
                req_char,
            }
        }

        fn get_opt_test_case(req_line: u64, req_char: u64) -> GotoDefintion {
            get_go_to_test_case(opt_go_to, req_line, req_char)
        }

        fn get_res_test_case(req_line: u64, req_char: u64) -> GotoDefintion {
            get_go_to_test_case(res_go_to, req_line, req_char)
        }

        vec![
            // option.sw
            get_opt_test_case(15, 12),
            get_opt_test_case(16, 17),
            get_opt_test_case(17, 29),
            get_opt_test_case(18, 19),
            get_opt_test_case(20, 13),
            get_opt_test_case(21, 19),
            get_opt_test_case(22, 29),
            get_opt_test_case(23, 18),
            // result.sw 
            get_res_test_case(20, 19),
            get_res_test_case(21, 25),
            get_res_test_case(22, 36),
            get_res_test_case(23, 27),
        ]
        .into_iter()
        .for_each(|(i, test_case)| {
            let _ = definition_check(&mut service, &go_to, i + 1).await;
        });
```

**Review by sdankel — DISMISSED** [review]
Nice one!

File: sway-lsp/src/server.rs [hunk 2]
```diff
@@ -807,103 +818,33 @@ mod tests {
...
+                    "line": go_to.def_line,
+                },
+                "start": {
+                    "character": go_to.def_start_char,
+                    "line": go_to.def_line,
+                }
+            });
+            assert_json_eq!(response.range, range);
+            assert!(uri.ends_with(go_to.def_path));
+        }
```
[the remark this answers is not in the document]
Comment (JoshuaBatty) [reply sway-lsp/src/server.rs:846]: sure that's a good idea.

**Review by mitchmindtree — DISMISSED** [review]

File: sway-lsp/src/server.rs (hunk 1 above)
Comment (JoshuaBatty) [reply sway-lsp/src/server.rs:1289]: Thanks, i've done something a little different but it looks a lot cleaner now.

**Review by sdankel — APPROVED** [review]

**Review by mitchmindtree — APPROVED** [review]

**Pull request merged**